### DISCUSSION TITLE
Incrementally hydrating editor state based on visible nodes

### DISCUSSION THREAD

**Discussion by runnedrun** [facebook/lexical#4740] [Q&A]
Hi! I have an application that stores fairly large documents (about 4000 lines of text). In order to make things smaller and faster I store each "line" as it's own entry in my DB that has it's own foreign key, and then I display each line as it's own LineNode in Lexical, which a foreign key pointing to the line in the DB. When saving my Lexical EditorState to my DB I replace each line with a simple reference to it's foreign key. Now, I don't want to hydrate all 4000 lines at once when I load the document the first time, instead I'd like to hydrate each LineNode with it's corresponding text, from my DB, when the node is scrolled into view. 

I have this kind of working right now, but it feels quite hacky. Are there any examples or plugins that can help with this? Thanks!

**Comment by CanRau** [comment]
Have you figured it out?
What's wrong with "just" appending the next line to the editorState? Not sure what you're doing but maybe you'd need to actually place something like `DecoratorNode`s as placeholders with the line.id and then replace with actual node and content once its time comes so that people can already start editing, maybe even make your placeholder nodes non-removable, ([this might help](https://github.com/facebook/lexical/discussions/4159))
